packages/f/ffmpeg/xmake.lua

```diff
@@ -45,6 +45,33 @@ package("ffmpeg")
         add_deps("yasm")
     end
 
+    if on_fetch then
+        on_fetch("mingw", "linux", "macosx", function (package, opt)
+            import("lib.detect.find_tool")
+            if opt.system then
+                local result
+                for _, name in ipairs({"libavcodec", "libavdevice", "libavfilter", "libavformat", "libavutil", "libpostproc", "libswresample", "libswscale"}) do
+                    local pkginfo = package:find_package("pkgconfig::" .. name, opt)
+                    if pkginfo then
+                        pkginfo.version = nil
+                        if not result then
+                            result = pkginfo
+                        else
+                            result = result .. pkginfo
+                        end
+                    else
+                        return {}
```

Comment:
Ok for nil but if the `else` is removed you could end up with a partially filled `result` (because some libs would be installed and some not) that will be considered valid, if I understand the code correctly.